src/utils/validate-template.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  NHSAppTemplate,
+  EmailTemplate,
+  SMSTemplate,
+  ChannelTemplate,
+  Template,
+} from '@utils/types';
+import { logger } from '@utils/logger';
+import {
+  $NHSAppTemplate,
+  $SMSTemplate,
+  $EmailTemplate,
+  $ChannelTemplate,
+} from './zod-validators';
+
+export const validateNHSAppTemplate = (
+  template: Template | undefined
+): NHSAppTemplate | undefined => {
+  try {
+    return $NHSAppTemplate.parse(template);
+  } catch (error) {
+    logger.error(error);
+    return undefined;
+  }
+};
+
+export const validateSMSTemplate = (
+  template: Template | undefined
+): SMSTemplate | undefined => {
+  try {
+    return $SMSTemplate.parse(template);
+  } catch (error) {
+    logger.error(error);
+    return undefined;
+  }
+};
+
+export const validateEmailTemplate = (
+  template: Template | undefined
+): EmailTemplate | undefined => {
+  try {
+    return $EmailTemplate.parse(template);
+  } catch (error) {
+    logger.error(error);
+    return undefined;
+  }
+};
+
+export const validateChannelTemplate = (
```

Comment:
You could probably just use this function instead of the template specific validateEmail/Sms/LetterTemplates function.

The `discriminatedUnion` should decide on which Zod schema to utilize to validate.

Reply:
AH yeah fair point! If you used `ChannelType` on any of the review pages a valid NHSAPP template would pass validation on an SMS review page.